src/pages/creations/LogosCarousel/index.tsx

```diff
@@ -0,0 +1,301 @@
+import { styled } from 'styled-components';
+import { MOBILE_MAX_WIDTH, TABLET_MAX_WIDTH } from '~/components';
+import { Carousel } from 'react-responsive-carousel';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { isMobile, isTablet } from '~/utils';
+
+interface LogosCarouselProps {
+  companies: string[];
+  onChange?: (index: number) => void;
+}
+
+const carouselState = {
+  desktop: {
+    loop: 7,
+    visibleItems: 9,
+  },
+  tablet: {
+    loop: 5,
+    visibleItems: 5,
+  },
+  mobile: {
+    loop: 5,
+    visibleItems: 3,
+  },
+};
+
+const formatCompanyToFileName = (company: string) => company.toLowerCase().replace(/ /g, '-');
+
+export default function LogosCarousel({ companies, onChange }: LogosCarouselProps) {
+  const [loop, setLoop] = useState(carouselState.desktop.loop); // iterations of the logos
+  const [visibleItems, setVisibleItems] = useState(carouselState.desktop.visibleItems); // number of logos visible
```

Comment:
I recommend avoiding commenting on every line and writing self-explanatory code.
loop state could be renamed to `interations`, or `logoIteration`.

`visibleItems`: it doesn’t need comments; it explains itself.
`maxSize`: the same, could be renamed to logosMaxSize.

Reply:
https://github.com/defi-wonderland/web/pull/114/commits/bdf96f491f213c63ab3ba85c7e8f07cfdff5f739